core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -646,7 +647,47 @@ private[spark] class AppStatusListener(
   }
 
   override def onUnpersistRDD(event: SparkListenerUnpersistRDD): Unit = {
-    liveRDDs.remove(event.rddId)
+    liveRDDs.remove(event.rddId).foreach { liveRDD =>
+      val executorsToUpdate = new HashSet[LiveExecutor]()
```

Comment:
I'd just call `maybeUpdate` directly when updating the executor. It's cheaper than inserting into a set, since a duplicate call will just compare the timestamp of the last update and then do nothing.

Reply:
An executor can be updated first when processing distributions and later when processing partitions. We want a duplicate call to do update in fact. If we just call `maybeUpdate` directly, the duplicate call will do nothing and miss updates from processing partitions.

Reply:
Right. But it would be nice to avoid the hash set if possible. The less stuff listeners have to do, the better.

Reply:
Ok. Good point for me. I updated the change to remove the hash set now.